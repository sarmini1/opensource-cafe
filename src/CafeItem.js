import React from "react";
import { Navigate, useParams } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";

/**CafeItem
 *
 * Props:
 * - items: array of objects [{id, name, description, recipe, serve}...]
 * - cantFind: string
 *
 * State:
 * - none
 *
 * App --> NavBar, Routes --> CafeItem
 */
function CafeItem({ items, cantFind }) {
  const { id } = useParams();

  let item = items.find(item => item.id === id);
  if (!item) return <Navigate to={cantFind} />;

  return (
    <section>
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {item.name}
          </CardTitle>
          <CardText className="font-italic">{item.description}</CardText>
          <p>
            <b>Recipe:</b> {item.recipe}
          </p>
          <p>
            <b>Serve:</b> {item.serve}
          </p>
        </CardBody>
      </Card>
    </section>
  );
}

export default CafeItem;