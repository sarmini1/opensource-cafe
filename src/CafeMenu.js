import React from "react";
import { Link } from "react-router-dom";
import "./CafeMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

/**CafeMenu
 *
 * Props:
 * - type: string
 * - items: array of objects [{id, name, description, recipe, serve}...]
 * - title: string
 *
 * State:
 * - none
 *
 * App --> NavBar, Routes --> CafeMenu
 */
function CafeMenu({ type, items, title }) {

  return (
    <section className="CafeMenu col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {title} Menu
          </CardTitle>
          <CardText>
            Please enjoy some of our {type} listed below!
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default CafeMenu;