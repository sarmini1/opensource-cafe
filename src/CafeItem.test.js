import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from "react-router-dom";
import CafeItem from "./CafeItem";
import CafeMenu from './CafeMenu';

const testSnacks = [
  {
    "id": "nachos",
    "name": "Nachos",
    "description": "An American classic!",
    "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
    "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
  },
  {
    "id": "hummus",
    "name": "Hummus",
    "description": "Sure to impress your vegan friends!",
    "recipe": "Purchase one container of hummus.",
    "serve": "Place unceremoniously on the table, along with pita bread."
  }
];

it('mounts without crashing', function () {
  render(
    <MemoryRouter>
      <CafeItem items={testSnacks} cantFind={"/snacks"} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { container } = render(
    <MemoryRouter>
      <CafeItem items={testSnacks} cantFind={"/snacks"} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

it("renders appropriate item content from URL parameter", function () {
  const { container } = render(
    <MemoryRouter initialEntries={["/snacks/nachos"]}>
      <Routes>
        <Route path="/snacks/:id" element={
          <CafeItem
            items={testSnacks}
            cantFind={('/snacks')}
          />}
        >
        </Route>
      </Routes>
    </MemoryRouter>
  );

  expect(container).toContainHTML("Cover expensive, organic tortilla chips with Cheez Whiz.");
});

it("navigates back to snacks page when item can't be found", function () {
  const { container } = render(
    <MemoryRouter initialEntries={["/snacks/fake-snack"]}>
      <Routes>
        <Route
          path="/snacks/:id"
          element={
            <CafeItem
              items={testSnacks}
              cantFind={('/snacks')}
            />}
        >
        </Route>
        <Route
          path="/snacks"
          element={
            <CafeMenu
              type="snacks"
              items={testSnacks}
              title="Snacks"
            />}
        />
      </Routes>
    </MemoryRouter>
  );

  expect(container).toContainHTML('Please enjoy some of our snacks listed below!');
  expect(container).not.toContainHTML('CafeItem');
});