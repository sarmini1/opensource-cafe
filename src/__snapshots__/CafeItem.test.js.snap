// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="font-weight-bold text-center card-title"
        >
           Menu
        </div>
        <p
          class="card-text"
        >
          Please enjoy some of our 
           listed below!
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/undefined/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/undefined/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
